Hardware/Sources/Hardware/StripeCardReader/StripeCardReaderService.swift

```diff
@@ -0,0 +1,116 @@
+import Combine
+import StripeTerminal
+
+/// The adapter wrapping the Stripe Terminal SDK
+public final class StripeCardReaderService {
+    private let tokenProvider: ConnectionTokenProvider
+
+    private let discoveredReadersSubject = CurrentValueSubject<[CardReader], Never>([])
+    private let connectedReadersSubject = PassthroughSubject<[CardReader], Never>()
```

Comment:
It might be possible for a view to be subscribing to the connectedReaders after a reader has been connected by the service. I think this means the connectedReadersSubject should be a CurrentValueSubject so the view can obtain initial state.